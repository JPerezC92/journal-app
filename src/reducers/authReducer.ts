import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export interface AuthState {
  isLoggedIn: boolean;
  displayName: string;
  uuid: string;
}

const initialState: AuthState = {
  isLoggedIn: false,
} as AuthState;

export const authSlice = createSlice({
  name: "[Auth]",
  initialState,
  reducers: {
    login: (_, action: PayloadAction<AuthState>): AuthState => action.payload,
    logout: () => initialState,
  },
});

export const { login, logout } = authSlice.actions;

const authReducer = authSlice.reducer;

export default authReducer;
