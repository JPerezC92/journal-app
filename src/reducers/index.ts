export * from "./authReducer";
export * from "./uiReducer";
export * from "./notesReducer";
