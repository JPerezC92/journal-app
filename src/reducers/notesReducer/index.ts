export * from "./notesReducer";
export * from "./notesThunks";
