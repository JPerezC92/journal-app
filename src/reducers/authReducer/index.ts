import authReducer from "./authReducer";

export * from "./authReducer";
export * from "./authReducerTypes";
export * from "./authThunks";

export default authReducer;
