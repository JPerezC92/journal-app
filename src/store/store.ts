import { configureStore } from "@reduxjs/toolkit";
import { uiReducer, authReducer, notesReducer } from "../reducers";

export const reducers = {
  authReducer,
  uiReducer,
  notesReducer,
};

export const store = configureStore({
  reducer: reducers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
