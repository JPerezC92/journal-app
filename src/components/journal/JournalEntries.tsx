import { useSelector } from "react-redux";
import { RootState } from "../../store/store";
import JournalEntry from "./JournalEntry";

const JournalEntries = () => {
  const { notes } = useSelector((state: RootState) => state.notesReducer);

  return (
    <div className="journal__entries">
      {notes.map((note) => (
        <JournalEntry key={note.id} note={note} />
      ))}
    </div>
  );
};

export default JournalEntries;
