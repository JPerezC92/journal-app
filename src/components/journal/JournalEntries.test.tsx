import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { Note, notesActions } from "../../reducers";
import { store } from "../../store/store";
import JournalEntries from "./JournalEntries";

const notesMock: Note[] = [
  {
    id: "1",
    title: "Test title 1",
    body: "Test body 1",
    date: 1626208415067,
    imageUrl:
      "https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png",
  },
  {
    id: "2",
    title: "Test title 2",
    body: "Test body 2",
    date: 1626989415067,
    imageUrl:
      "https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png",
  },
];

describe("Test on <JournalEntries />", () => {
  test("should match the snapshot correctly", () => {
    store.dispatch(notesActions.setNotes(notesMock));
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <JournalEntries />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
