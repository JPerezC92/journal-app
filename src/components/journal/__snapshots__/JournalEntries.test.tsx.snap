// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <JournalEntries /> should match the snapshot correctly 1`] = `
<div
  className="journal__entries"
>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="journal__entry-picture"
      role="img"
      style={
        Object {
          "backgroundImage": "url(https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png)",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Test title 1
      </p>
      <p
        className="journal__entry-content"
      >
        Test body 1
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Tuesday
      </span>
      <h4>
        13th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="journal__entry-picture"
      role="img"
      style={
        Object {
          "backgroundImage": "url(https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png)",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Test title 2
      </p>
      <p
        className="journal__entry-content"
      >
        Test body 2
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Thursday
      </span>
      <h4>
        22nd
      </h4>
    </div>
  </div>
</div>
`;
